asdf/asdf.py

```diff
@@ -116,6 +117,14 @@ def __init__(
             When `False`, when reading files, attempt to memmap underlying data
             arrays when possible.
 
+        memmap_arrays : bool, optional
```

Comment:
On second thought, should we just make this `memmap`?

Reply:
I updated this PR to `memmap`. I'm happy to dicuss altenatives (like `memmap_blocks`) and don't feel strongly about the naming of this argument. I went with `memmap` for a few reasons:
- it's shorter to type than `memmap_blocks` which still describing the basic option (memory mapping)
- in case we come up with some new crazy way to memory map things (like the tree) the option should be forward compatible